### PR TITLE
[iOS/macOS] Add option to automatically build (and sign / archive) bundles.

### LINKED ISSUE #1460: Create an .app bundle automatically when compiling the editor for macOS
*See also https://github.com/godotengine/godot/issues/16415.*

**Describe the project you are working on:**

The Godot editor :slightly_smiling_face: 

**Describe the problem or limitation you are having in your project:**

When building the editor on macOS, you won't get any system integration out of the box. To get Godot to look like an usual macOS application (that can be dragged to `/Applications` and the like), you have to place the binary in [`misc/dist/osx_tools.app`](https://github.com/godotengine/godot/tree/master/misc/dist/osx_tools.app) yourself in the correct location (`Contents/MacOS/Godot`).

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

We can have SCons generate the .app bundle automatically when building the editor for macOS.

Users will still be able to start the application from a terminal by running `bin/<app_bundle_name>.app/Contents/MacOS/Godot`.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

Make SCons generate the .app bundle automatically by using its file handling utilities. To do so, it can copy the files contained in [`misc/dist/osx_tools.app`](https://github.com/godotengine/godot/tree/master/misc/dist/osx_tools.app) to the `bin` folder and move the compiled Godot binary to the correct location.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

Technically, yes: this can be achieved with a few lines of shell scripting. However, since this is about improving the experience for people who are just getting started with compiling the engine, I'd say it's worth the effort (provided we don't find downsides to doing this).

If there are significant downsides to doing this, we can put this behind a SCons option like `osx_app_bundle=yes`.

**Is there a reason why this should be core and not an add-on in the asset library?:**

This is core buildsystem functionality.

### PR DESCRIPTION
**Pull request by bruvzg**
- Adds `generate_bundle` build flag to automatically generate .app bundle or template zip.
- Adds `bundle_sign_identity` to automatically sign editor .app bundle with correct entitlements (defaults to ad-hoc signing is not set).

```
scons platform=macos target=editor arch=x86_64
scons platform=macos target=editor arch=arm64 generate_bundle=yes
# generates and signs editor app bundle with universal executable.
```

```
scons platform=macos target=template_release arch=x86_64
scons platform=macos target=template_release arch=arm64
scons platform=macos target=template_debug arch=x86_64
scons platform=macos target=template_debug arch=arm64 generate_bundle=yes
# generates ZIP with the full macOS export template (debug and release universal executable).
```

```
scons platform=ios target=template_release arch=arm64
scons platform=ios target=template_release arch=x86_64 ios_simulator=yes
scons platform=ios target=template_release arch=arm64 ios_simulator=yes
scons platform=ios target=template_debug arch=arm64
scons platform=ios target=template_debug arch=x86_64 ios_simulator=yes
scons platform=ios target=template_debug arch=arm64 ios_simulator=yes generate_bundle=yes
# generates ZIP with the full iOS export template (debug and release universal xcframeworks for device and simulator).
```

Fixes https://github.com/godotengine/godot-proposals/issues/1460

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -183,6 +183,7 @@ opts.Add(BoolVariable("separate_debug_symbols", "Extract debugging symbols to a
 opts.Add(EnumVariable("lto", "Link-time optimization (production builds)", "none", ("none", "auto", "thin", "full")))
 opts.Add(BoolVariable("production", "Set defaults to build Godot for use in production", False))
 opts.Add(BoolVariable("generate_apk", "Generate an APK/AAB after building Android library by calling Gradle", False))
+opts.Add(BoolVariable("generate_bundle", "Generate an APP bundle after building iOS/macOS binaries", False))
```
Comment (akien-mga) [inline SConstruct:186]: These could maybe be added in each platform's `detect.py` (`get_opts`).

This also applies to `generate_apk`, I missed it when reviewing that PR. CC @Calinou

Comment (akien-mga) [reply SConstruct:186]: WDYT @bruvzg?

**Review by akien-mga — APPROVED** [review]
Looks good to me overall.

File: SConstruct (hunk 1 above)
Comment (bruvzg) [reply SConstruct]: Rebased and move both options to the platforms.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by zaddok** [comment]
When I follow these instructions, we get a godot_ios_simulator.zip file. When trying to use Godot's menu "Manage Export Templates" it fails to import:

image

Is there another missing step?

**Pull request merged**